### ISSUE TITLE
Concurrent Map-Access

### ISSUE THREAD

**Issue by MaZderMind** [MaZderMind/live-hls-transcode#6]
Concurrent write access on last access time update

**Comment by MaZderMind** [comment]
seems to still happen
```
concurrent map writes

goroutine 3946 [running]:
runtime.throw(0x911105, 0x15)
	/usr/lib/go/src/runtime/panic.go:774 +0x72 fp=0xc00039b5e8 sp=0xc00039b5b8 pc=0x42f612
runtime.mapassign_faststr(0x875160, 0xc0003f1c80, 0xc00030e240, 0x3e, 0xc000496788)
	/usr/lib/go/src/runtime/map_faststr.go:291 +0x3fe fp=0xc00039b650 sp=0xc00039b5e8 pc=0x414b6e
main.StreamStatusManager.UpdateLastAccess(0x9120d5, 0x17, 0xc0003f1c80, 0x0, 0x0, 0x0, 0x3e8, 0xc00030e240, 0x3e)
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/stream_status_manager.go:185 +0x1ec fp=0xc00039b778 sp=0xc00039b650 pc=0x80b2dc
main.StreamHandler.HandlePlaylistRequest(0xc0003c66c0, 0xda2ba0, 0xc000346000, 0xc000384100, 0xc00030e240, 0x3e, 0xc00030e080, 0x38, 0xc00030e27b, 0x3, ...)
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/stream_handler.go:25 +0xbd fp=0xc00039ba10 sp=0xc00039b778 pc=0x80787d
main.main.func1(0xda2ba0, 0xc000346000, 0xc000384100)
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/main.go:45 +0x45a fp=0xc00039bb90 sp=0xc00039ba10 pc=0x80db2a
net/http.HandlerFunc.ServeHTTP(0xc0003a7000, 0xda2ba0, 0xc000346000, 0xc000384100)
	/usr/lib/go/src/net/http/server.go:2007 +0x44 fp=0xc00039bbb8 sp=0xc00039bb90 pc=0x702e54
net/http.(*ServeMux).ServeHTTP(0x10b09c0, 0xda2ba0, 0xc000346000, 0xc000384100)
	/usr/lib/go/src/net/http/server.go:2387 +0x1bd fp=0xc00039bc18 sp=0xc00039bbb8 pc=0x704e2d
net/http.serverHandler.ServeHTTP(0xc000456000, 0xda2ba0, 0xc000346000, 0xc000384100)
	/usr/lib/go/src/net/http/server.go:2802 +0xa4 fp=0xc00039bc48 sp=0xc00039bc18 pc=0x7063a4
net/http.(*conn).serve(0xc0000c61e0, 0xda3ae0, 0xc000032040)
	/usr/lib/go/src/net/http/server.go:1890 +0x875 fp=0xc00039bfc8 sp=0xc00039bc48 pc=0x701c45
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc00039bfd0 sp=0xc00039bfc8 pc=0x45be41
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2927 +0x38e

goroutine 1 [IO wait]:
internal/poll.runtime_pollWait(0x7f6ac5ae20a8, 0x72, 0x0)
	/usr/lib/go/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc0003a7118, 0x72, 0x0, 0x0, 0x9083da)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Accept(0xc0003a7100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:384 +0x1f8
net.(*netFD).accept(0xc0003a7100, 0xc00053b9f8, 0x707354, 0xc0004560a0)
	/usr/lib/go/src/net/fd_unix.go:238 +0x42
net.(*TCPListener).accept(0xc0003e8fe0, 0x5d99c91a, 0xc00053b9f8, 0x4c3c26)
	/usr/lib/go/src/net/tcpsock_posix.go:139 +0x32
net.(*TCPListener).Accept(0xc0003e8fe0, 0xc00053ba48, 0x18, 0xc000000180, 0x70686e)
	/usr/lib/go/src/net/tcpsock.go:261 +0x47
net/http.(*Server).Serve(0xc000456000, 0xda28e0, 0xc0003e8fe0, 0x0, 0x0)
	/usr/lib/go/src/net/http/server.go:2896 +0x286
net/http.(*Server).ListenAndServe(0xc000456000, 0xc000456000, 0x0)
	/usr/lib/go/src/net/http/server.go:2825 +0xb7
net/http.ListenAndServe(...)
	/usr/lib/go/src/net/http/server.go:3080
main.main()
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/main.go:54 +0x64e

goroutine 23 [IO wait]:
internal/poll.runtime_pollWait(0x7f6ac5ae1fd8, 0x72, 0xffffffffffffffff)
	/usr/lib/go/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc000458018, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc000458000, 0xc000242000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:169 +0x1cf
net.(*netFD).Read(0xc000458000, 0xc000242000, 0x1000, 0x1000, 0xc000039d70, 0xc000135878, 0x40b0ba)
	/usr/lib/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc000140008, 0xc000242000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:184 +0x68
net/http.(*connReader).Read(0xc000454000, 0xc000242000, 0x1000, 0x1000, 0x7f6ac5a9bec8, 0x0, 0x800203000)
	/usr/lib/go/src/net/http/server.go:785 +0xf4
bufio.(*Reader).fill(0xc00007e420)
	/usr/lib/go/src/bufio/bufio.go:100 +0x103
bufio.(*Reader).ReadSlice(0xc00007e420, 0xa, 0x7f6ac5a9d228, 0xc0001359a8, 0x40dab6, 0xc0001b0000, 0x100)
	/usr/lib/go/src/bufio/bufio.go:359 +0x3d
bufio.(*Reader).ReadLine(0xc00007e420, 0xc0001359b0, 0xc0004a4380, 0x7f6ac5b1b8b8, 0x0, 0x40e328, 0x30)
	/usr/lib/go/src/bufio/bufio.go:388 +0x34
net/textproto.(*Reader).readLineSlice(0xc000454060, 0xc0001b0000, 0xc000458000, 0x0, 0x0, 0x43143c)
	/usr/lib/go/src/net/textproto/reader.go:57 +0x6c
net/textproto.(*Reader).ReadLine(...)
	/usr/lib/go/src/net/textproto/reader.go:38
net/http.readRequest(0xc00007e420, 0x0, 0xc0001b0000, 0x0, 0x0)
	/usr/lib/go/src/net/http/request.go:1012 +0x92
net/http.(*conn).readRequest(0xc00045c000, 0xda3ae0, 0xc000462000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/http/server.go:965 +0x15f
net/http.(*conn).serve(0xc00045c000, 0xda3ae0, 0xc000462000)
	/usr/lib/go/src/net/http/server.go:1817 +0x6d4
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2927 +0x38e

goroutine 10 [select, 1 minutes]:
main.Cleanup.Start.func1(0xc0003b87d0, 0xc00002f320, 0xc0003c66c0, 0xc0000005a0)
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/cleanup.go:29 +0xdb
created by main.Cleanup.Start
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/cleanup.go:27 +0x5b

goroutine 31 [IO wait]:
internal/poll.runtime_pollWait(0x7f6ac5ae1f08, 0x72, 0xffffffffffffffff)
	/usr/lib/go/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc0003a6418, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0003a6400, 0xc0001b4000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:169 +0x1cf
net.(*netFD).Read(0xc0003a6400, 0xc0001b4000, 0x1000, 0x1000, 0x30, 0x38, 0x7f6ac5ae32f0)
	/usr/lib/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc00031c030, 0xc0001b4000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:184 +0x68
net/http.(*connReader).Read(0xc000455ad0, 0xc0001b4000, 0x1000, 0x1000, 0xc000322180, 0x0, 0x800000000)
	/usr/lib/go/src/net/http/server.go:785 +0xf4
bufio.(*Reader).fill(0xc00007ede0)
	/usr/lib/go/src/bufio/bufio.go:100 +0x103
bufio.(*Reader).ReadSlice(0xc00007ede0, 0xc000348d0a, 0x7f6ac5ae30d0, 0xc0001299a8, 0x40dab6, 0xc000360400, 0x100)
	/usr/lib/go/src/bufio/bufio.go:359 +0x3d
bufio.(*Reader).ReadLine(0xc00007ede0, 0xc0001299b0, 0xc0004a4000, 0x7f6ac5b19d98, 0x0, 0x8e5100, 0xc0003f1da0)
	/usr/lib/go/src/bufio/bufio.go:388 +0x34
net/textproto.(*Reader).readLineSlice(0xc0003f1da0, 0xc000360400, 0xc0003a6400, 0x0, 0x0, 0x43143c)
	/usr/lib/go/src/net/textproto/reader.go:57 +0x6c
net/textproto.(*Reader).ReadLine(...)
	/usr/lib/go/src/net/textproto/reader.go:38
net/http.readRequest(0xc00007ede0, 0x0, 0xc000360400, 0x0, 0x0)
	/usr/lib/go/src/net/http/request.go:1012 +0x92
net/http.(*conn).readRequest(0xc0000c7720, 0xda3ae0, 0xc0003c6dc0, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/http/server.go:965 +0x15f
net/http.(*conn).serve(0xc0000c7720, 0xda3ae0, 0xc0003c6dc0)
	/usr/lib/go/src/net/http/server.go:1817 +0x6d4
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2927 +0x38e

goroutine 43 [IO wait]:
internal/poll.runtime_pollWait(0x7f6ac5ae1e38, 0x72, 0xffffffffffffffff)
	/usr/lib/go/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc00018e418, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc00018e400, 0xc000188000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:169 +0x1cf
net.(*netFD).Read(0xc00018e400, 0xc000188000, 0x1000, 0x1000, 0x0, 0x3030320000000000, 0xed52bc017)
	/usr/lib/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc000010060, 0xc000188000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:184 +0x68
net/http.(*connReader).Read(0xc0003f1e60, 0xc000188000, 0x1000, 0x1000, 0xc0003c6380, 0x10672be, 0x2)
	/usr/lib/go/src/net/http/server.go:785 +0xf4
bufio.(*Reader).fill(0xc00007ed20)
	/usr/lib/go/src/bufio/bufio.go:100 +0x103
bufio.(*Reader).ReadSlice(0xc00007ed20, 0xa, 0xc000135b28, 0xc0001359a8, 0x40dab6, 0xc000360500, 0x100)
	/usr/lib/go/src/bufio/bufio.go:359 +0x3d
bufio.(*Reader).ReadLine(0xc00007ed20, 0xc0001359b0, 0xc000488000, 0x7f6ac5b19d98, 0x0, 0x2, 0xc0001359f0)
	/usr/lib/go/src/bufio/bufio.go:388 +0x34
net/textproto.(*Reader).readLineSlice(0xc0000a2780, 0xc000360500, 0xc00018e400, 0x0, 0x0, 0x43143c)
	/usr/lib/go/src/net/textproto/reader.go:57 +0x6c
net/textproto.(*Reader).ReadLine(...)
	/usr/lib/go/src/net/textproto/reader.go:38
net/http.readRequest(0xc00007ed20, 0x0, 0xc000360500, 0x0, 0x0)
	/usr/lib/go/src/net/http/request.go:1012 +0x92
net/http.(*conn).readRequest(0xc00036c000, 0xda3ae0, 0xc0000e2640, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/http/server.go:965 +0x15f
net/http.(*conn).serve(0xc00036c000, 0xda3ae0, 0xc0000e2640)
	/usr/lib/go/src/net/http/server.go:1817 +0x6d4
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2927 +0x38e

goroutine 44 [IO wait]:
internal/poll.runtime_pollWait(0x7f6ac5ae1bc8, 0x72, 0xffffffffffffffff)
	/usr/lib/go/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc00018e498, 0x72, 0x1000, 0x1000, 0xffffffffffffffff)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc00018e480, 0xc00036e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:169 +0x1cf
net.(*netFD).Read(0xc00018e480, 0xc00036e000, 0x1000, 0x1000, 0x0, 0x3030320000000000, 0xed52bbffe)
	/usr/lib/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc000010068, 0xc00036e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:184 +0x68
net/http.(*connReader).Read(0xc00045aa50, 0xc00036e000, 0x1000, 0x1000, 0xc000462180, 0x10672be, 0x203000)
	/usr/lib/go/src/net/http/server.go:785 +0xf4
bufio.(*Reader).fill(0xc0003680c0)
	/usr/lib/go/src/bufio/bufio.go:100 +0x103
bufio.(*Reader).ReadSlice(0xc0003680c0, 0xa, 0x1, 0xc00039f9a8, 0x40dca4, 0xc0001b0100, 0x100)
	/usr/lib/go/src/bufio/bufio.go:359 +0x3d
bufio.(*Reader).ReadLine(0xc0003680c0, 0xc00039f9b0, 0xc000496000, 0x7f6ac5b1b8b8, 0xc000464780, 0x40e328, 0x30)
	/usr/lib/go/src/bufio/bufio.go:388 +0x34
net/textproto.(*Reader).readLineSlice(0xc000454600, 0xc0001b0100, 0xc00018e480, 0x0, 0x0, 0xd8a944)
	/usr/lib/go/src/net/textproto/reader.go:57 +0x6c
net/textproto.(*Reader).ReadLine(...)
	/usr/lib/go/src/net/textproto/reader.go:38
net/http.readRequest(0xc0003680c0, 0x0, 0xc0001b0100, 0x0, 0x0)
	/usr/lib/go/src/net/http/request.go:1012 +0x92
net/http.(*conn).readRequest(0xc00036c0a0, 0xda3ae0, 0xc000462140, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/http/server.go:965 +0x15f
net/http.(*conn).serve(0xc00036c0a0, 0xda3ae0, 0xc000462140)
	/usr/lib/go/src/net/http/server.go:1817 +0x6d4
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2927 +0x38e

goroutine 3960 [IO wait]:
internal/poll.runtime_pollWait(0x7f6ac5ae1d68, 0x72, 0xffffffffffffffff)
	/usr/lib/go/src/runtime/netpoll.go:184 +0x55
internal/poll.(*pollDesc).wait(0xc0003a6018, 0x72, 0x0, 0x1, 0xffffffffffffffff)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/lib/go/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Read(0xc0003a6000, 0xc0005260d1, 0x1, 0x1, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:169 +0x1cf
net.(*netFD).Read(0xc0003a6000, 0xc0005260d1, 0x1, 0x1, 0xc000463d98, 0xc00038af68, 0x467fdc)
	/usr/lib/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc00031c010, 0xc0005260d1, 0x1, 0x1, 0x0, 0x0, 0x0)
	/usr/lib/go/src/net/net.go:184 +0x68
net/http.(*connReader).backgroundRead(0xc0005260c0)
	/usr/lib/go/src/net/http/server.go:677 +0x58
created by net/http.(*connReader).startBackgroundRead
	/usr/lib/go/src/net/http/server.go:673 +0xd4

goroutine 3926 [runnable]:
net/http.(*connReader).backgroundRead(0xc0000200f0)
	/usr/lib/go/src/net/http/server.go:676
created by net/http.(*connReader).startBackgroundRead
	/usr/lib/go/src/net/http/server.go:673 +0xd4

goroutine 3959 [runnable]:
path/filepath.(*lazybuf).string(...)
	/usr/lib/go/src/path/filepath/path.go:57
path/filepath.Clean(0xc0002fc150, 0x69, 0x2, 0x9056a6)
	/usr/lib/go/src/path/filepath/path.go:159 +0x929
path/filepath.join(0xc0003995d0, 0x2, 0x2, 0xa, 0xa)
	/usr/lib/go/src/path/filepath/path_unix.go:45 +0xbf
path/filepath.Join(...)
	/usr/lib/go/src/path/filepath/path.go:210
net/http.Dir.Open(0xc00002e5a0, 0x5e, 0xc0001d8220, 0xa, 0x9, 0xc000399678, 0x4e1b9a, 0xc00056e040)
	/usr/lib/go/src/net/http/fs.go:76 +0xe4
net/http.serveFile(0xda2ba0, 0xc0004560e0, 0xc0004b4100, 0xd9a2a0, 0xc000174230, 0xc0001d8220, 0xa, 0x892601)
	/usr/lib/go/src/net/http/fs.go:553 +0xb0
net/http.(*fileHandler).ServeHTTP(0xc000174250, 0xda2ba0, 0xc0004560e0, 0xc0004b4100)
	/usr/lib/go/src/net/http/fs.go:725 +0xa1
main.StreamHandler.HandleSegmentRequest(0xc0003c66c0, 0xda2ba0, 0xc0004560e0, 0xc0004b4100, 0xc000026300, 0x3e, 0xc000026140, 0x38, 0xc00002633b, 0x3, ...)
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/stream_handler.go:90 +0x49e
main.main.func1(0xda2ba0, 0xc0004560e0, 0xc0004b4100)
	/home/peter/go/src/github.com/mazdermind/live-hls-transcode/main.go:47 +0x3fd
net/http.HandlerFunc.ServeHTTP(0xc0003a7000, 0xda2ba0, 0xc0004560e0, 0xc0004b4100)
	/usr/lib/go/src/net/http/server.go:2007 +0x44
net/http.(*ServeMux).ServeHTTP(0x10b09c0, 0xda2ba0, 0xc0004560e0, 0xc0004b4100)
	/usr/lib/go/src/net/http/server.go:2387 +0x1bd
net/http.serverHandler.ServeHTTP(0xc000456000, 0xda2ba0, 0xc0004560e0, 0xc0004b4100)
	/usr/lib/go/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc00045c0a0, 0xda3ae0, 0xc0000e2040)
	/usr/lib/go/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2927 +0x38e
```

**Comment by MaZderMind** [comment]
fixed with 7384838… I hope
